packages/fast-components-react-msft/src/action-toggle/action-toggle.spec.tsx

```diff
@@ -0,0 +1,291 @@
+        actionToggle__disabled: "action-toggle-disabled",
+        actionToggle__selected: "action-toggle-selected",
+        actionToggle__hasGlyphAndContent: "action-toggle-hasGlyphAndContent",
+    };
+    const href: string = "#";
+
+    test("should have a displayName that matches the component name", () => {
+        expect((MSFTActionToggle as any).name).toBe(MSFTActionToggle.displayName);
+    });
+
+    test("should not throw if managedClasses are not provided", () => {
+        expect(() => {
+            shallow(
+                <MSFTActionToggle selectedLabel="selected" unselectedLabel="unselected" />
+            );
+        }).not.toThrow();
+    });
+
+    test("should create unselectedglyph element passed in", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            selectedGlyph: (className?: string): React.ReactNode => {
+                return <div>X</div>;
+            },
+            unselectedGlyph: (className?: string): React.ReactNode => {
+                return <div>Y</div>;
+            },
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.contains(<div>Y</div>)).toEqual(true);
+    });
+
+    test("should create selectedglyph element passed in", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            selectedGlyph: (className?: string): React.ReactNode => {
+                return <div>X</div>;
+            },
+            unselectedGlyph: (className?: string): React.ReactNode => {
+                return <div>Y</div>;
+            },
+            selected: true,
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.contains(<div>X</div>)).toEqual(true);
+    });
+
+    test("should implement unhandledProps", () => {
+        const handledProps: ActionToggleProps = {
+            managedClasses,
+            href,
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+        };
+
+        const unhandledProps: ActionToggleUnhandledProps = {
+            "aria-label": "label",
+        };
+
+        const props: ActionToggleProps = { ...handledProps, ...unhandledProps };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.first().prop("aria-label")).toEqual("label");
+    });
+
+    test("should apply a 'disabled' html class when prop disabled is passed", () => {
+        const props: ActionToggleHandledProps = {
+            selectedLabel: "selected",
+            unselectedLabel: "unselected",
+            disabled: true,
+        };
+
+        const rendered: any = mount(<ActionToggle {...props} />);
+
+        expect(rendered.find("button").prop("className")).toContain(
+            "actionToggle__disabled"
```

Comment:
I would prefer to interpolate this class name in case it changes i.e. `${managedClasses.actionToggle__disabled}`

Reply:
That does not actually return the string we're checking for.  Hardcoding the string being checked for seems to be a consistent pattern - search the repo for 'prop("className")).toContain('  and you get a bunch of hits.  Is there an example of a better way?  (I'm also realizing that the managedClasses declaration at the top might not serve any useful purpose)

Reply:
Managed classes declaration at the top is typically useful to remove duplication (not having to declare that object over and over and over for individual tests). As far as checking via accessing the managed classes const - can we write an issue to address this? I think it warrants discussion regarding cost/benefit and any potential gotcha's. I don't think it's necessary to block this PR on that change unless there is a clear problem being solved in this PR

Reply:
I forgot the `.` at the beginning, so that is why it probably didn't work. This can be done in another task to clean up usage, but I do feel like it is the more correct way, as per my reasoning.